### PR TITLE
Added witdh and height as parameters to procedural face

### PR DESCRIPTION
**Pull request by ca2-chambers**
Makes it possible to generate face of arbitrary size, for applications outside of the Cozmo robot.

### CODE REVIEW

File: pycozmo/procedural_face.py [hunk 1]
```diff
@@ -1,72 +1,68 @@
...
-
-EYE_WIDTH = 28
-EYE_HEIGHT = 40
-HALF_EYE_WIDTH = EYE_WIDTH // 2
-HALF_EYE_HEIGHT = EYE_HEIGHT // 2
+import math
 
 RESAMPLE = Image.NEAREST
 
+class ProceduralMaster(object):
```
[the remark this answers is not in the document]
Comment (zayfod) [reply pycozmo/procedural_face.py:8]: Let's call the class `ProceduralBase` instead.

**Review by zayfod — COMMENTED** [review]
I really like the idea.

The code looks good overall.

Please run it through `flake8` and verify that it adheres to PEP8.

**Comment by zayfod** [comment]
Changed the target branch to `dev`. `master` is used only for releases.

**Review by zayfod — APPROVED** [review]
I did minor clean up, tested with Cozmo, and like the end result.

If it works for you, I'd be happy to merge.

On a separate note, I really like what you have done with https://git.brl.ac.uk/ca2-chambers/expressive-eyes . Have you tried visualizing these faces on Cozmo? Would you be interested in integrating them as a new procedural face demo?

**Comment by ca2-chambers** [comment]
Dear Kaloyan,

I would be interested in integrating them as a demo, however, I no longer have access to the gitlab code as it is part of my dissertation. I will discuss this with my supervisor.

Thanks,

Catherine Chambers
________________________________

**Comment by zayfod** [comment]
PyCozmo is involved in your dissertation!? That's awesome! 🍺 

Good luck!

Please, keep me posted.

**Comment by ca2-chambers** [comment]
I asked my supervisor and he said that I can use it. What do you need me to do?

On 27 Sep 2020 4:51 pm, Kaloyan Tenchov <notifications@github.com> wrote:

PyCozmo is involved in your dissertation!? That's awesome! 🍺

Good luck!

Please, keep me posted.

**Comment by zayfod** [comment]
Being able to see all emotions from Expressive Eyes on Cozmo's face would be awesome.

I am imagining an example application that just keeps interpolating between all of them.

This could be the base for procedural face rendering in the new "brain" application.

**Pull request merged**